### ISSUE TITLE
Contained resources with longer ids

### ISSUE THREAD

**Issue by chempro** [FSHSchool/FSHOnline#116]
Firstly, great work on FSH! It makes it so much easier only having to worry about the 'differences' from the base profile

I'm sure the issue here is the operator, but I can't work out what I'm doing wrong

Using https://fshschool.org/FSHOnline/#/

Using the below example. Copy it into the righthand JSON side, then click 'Convert to FSH' and click 'Convert to JSON' and you get exactly what you started with which is great
```
{
  "resourceType": "Condition",
  "id": "EvesCondition",
  "contained": [
    {
      "resourceType": "Patient",
      "id": "f1",
      "name": [
        {
          "given": [
            "Eve"
          ],
          "family": "Anyperson"
        }
      ]
    }
  ],
  "subject": {
    "reference": "#f1"
  }
}
```

Now, try again but this time use a longer id e.g. a guid
Input
```
{
  "resourceType": "Condition",
  "id": "EvesCondition",
  "contained": [
    {
      "resourceType": "Patient",
      "id": "788a5d78-6353-45f7-93f6-45dc2371173c",
      "name": [
        {
          "given": [
            "Eve"
          ],
          "family": "Anyperson"
        }
      ]
    }
  ],
  "subject": {
    "reference": "#788a5d78-6353-45f7-93f6-45dc2371173c"
  }
}
```
click 'Convert to FSH' and click 'Convert to JSON'
Output
```
{
  "resourceType": "Condition",
  "id": "EvesCondition",
  "contained": [
    {
      "resourceType": "Patient",
      "id": "788a5d78-6353-45f7-93f6-45dc2371173c",
      "name": [
        {
          "given": [
            "Eve"
          ],
          "family": "Anyperson"
        }
      ]
    }
  ],
  "subject": {
    "reference": "788a5d78-6353-45f7-93f6-45dc2371173c"
  }
}
```
Note: How they not exactly the same now, subject.reference is missing the contained hash sign

**Comment by jafeltra** [comment]
Thanks for reporting this @chempro. I looked into this and there are a couple things going on here. GoFSH (which is run when you click the "Convert to FSH" button) is actually (incorrectly) interpreting the second id `788a5d78-6353-45f7-93f6-45dc2371173c` as a number and it does not use it with the `Instance` keyword and instead generates a name to put there. However, it does add the id `788a5d78-6353-45f7-93f6-45dc2371173c` to the instance with an assigned value rule. This produces the following FSH:

```
Instance: Inline-Instance-for-EvesCondition-1
InstanceOf: Patient
Usage: #inline
* id = "788a5d78-6353-45f7-93f6-45dc2371173c"
// ... a few other rules as well
```

While this is a little unexpected, it does produce the same output for the Instance, so the Instance that is generated is still technically a valid representation of the inline instance in your JSON.

The problem comes in when SUSHI (which is run when you click the "Convert to JSON" button) tries to reference the Patient instance in the Condition instance, in this FSH:

```
Instance: EvesCondition
InstanceOf: Condition
Usage: #example
* subject = Reference(788a5d78-6353-45f7-93f6-45dc2371173c)
// ... a few other rules as well
```

SUSHI has a problem when it tries to reference the line instance in the rule `* subject = Reference(788a5d78-6353-45f7-93f6-45dc2371173c)`. Because that id was set on a rule on the Patient instance instead of with the `Instance` keyword, SUSHI does not properly resolve the reference. This is what causes the hash sign to go missing after converting this example back to JSON. This is a bug in SUSHI that needs to be fixed. I've logged an issue on the SUSHI repo [here](https://github.com/FHIR/sushi/issues/1183). Once that issue is resolved, this issue should be resolved as well.

This doesn't happen in the first example you provided because the id `f1` does not get interpreted as a number in GoFSH, so the Instance that is created has `f1` assigned to the id via the `Instance` keyword, not a separate rule, and so SUSHI correctly generates the reference in the JSON with the hash sign.
